src/webgpu/api/operation/compute/overrides.spec.ts

```diff
@@ -0,0 +1,409 @@
+export const description = `
+Compute pipeline using overridable constants test.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { range } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+class F extends GPUTest {
+  ExpectShaderOutputWithConstants(
+    expected: Uint32Array | Float32Array,
+    constants: Record<string, GPUPipelineConstantValue>,
+    code: string
+  ) {
+    const dst = this.device.createBuffer({
+      size: expected.byteLength,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.STORAGE,
+    });
+
+    const pipeline = this.device.createComputePipeline({
+      layout: 'auto',
+      compute: {
+        module: this.device.createShaderModule({
+          code,
+        }),
+        entryPoint: 'main',
+        constants,
+      },
+    });
+
+    const bindGroup = this.device.createBindGroup({
+      entries: [{ binding: 0, resource: { buffer: dst, offset: 0, size: expected.byteLength } }],
+      layout: pipeline.getBindGroupLayout(0),
+    });
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginComputePass();
+    pass.setPipeline(pipeline);
+    pass.setBindGroup(0, bindGroup);
+    pass.dispatchWorkgroups(1);
+    pass.end();
+    this.device.queue.submit([encoder.finish()]);
+
+    this.expectGPUBufferValuesEqual(dst, expected);
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('basic')
+  .desc(`Test that correct constants override values or default values are used.`)
```

Comment:
I think all of the descriptions should have a little more detail on how the test works, so a reader can understand the coverage of the test better by reading the description.